### PR TITLE
Fix for config_section bug with subcommands

### PR DESCRIPTION
**Pull request by tbartlett0**
* Since pushing that bunch of changes didn't look to work so well, I'll just push my own change on its own for now... *

Reorder code in cmdline:create_parser to correct a bug where the config_section setting was being ignored for the main function when subcommands are used. (The subcommand setup is overwriting the config section used by populate_parser, so do it after setting up the main function.)

### CODE REVIEW

**Comment by sh0oki** [comment]
@tbartlett0 Thanks! 
Can you please check why the Python 2.6 test fails?

File: begin/cmdline.py [hunk 1]
```diff
@@ -171,6 +171,15 @@ def create_parser(func, env_prefix=None, config_file=None, config_section=None,
             description = func.__doc__,
             formatter_class=formatter_class
     )
+    # Main function
+    while hasattr(func, '__wrapped__') and not hasattr(func, '__signature__'):
+        if isinstance(func, extensions.Extension):
+            func.add_arguments(parser, defaults)
+            have_extensions = True
```
Comment (sh0oki) [inline begin/cmdline.py]: Is this variable still needed?

Comment (tbartlett0) [reply begin/cmdline.py]: It doesn't look like it, this and the other just below are the only uses I see in the entire code. Both assignments, and never used elsewhere.

**Comment by tbartlett0** [comment]
The Python 2.6 failure appears to be something to do with _coverage_:
```
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.6.9/bin/coverage", line 9, in <module>
    load_entry_point('coverage==4.3.2', 'console_scripts', 'coverage')()
< ... >
  File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/coverage/misc.py", line 287, in <module>
    class StopEverything(unittest.SkipTest):

AttributeError: 'module' object has no attribute 'SkipTest'
```
I don't _think_ anything I changed should have caused this? I will see if I can replicate it on my own box with python 2.6.

**Comment by tbartlett0** [comment]
I had a painful time trying to get python 2.6 to even install and work, but once I did I was able to run the tests without error. Is it possible that something in the test environment has changed since the last successful 2.6 build?

Update: from https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures
"25.3.6. Skipping tests and expected failures¶
_New in version 2.7._"
So unittest.SkipTest doesn't exist in python 2.6, but why is the test trying to use it and why did it not have this issue previously? (https://travis-ci.org/sh0oki/begins/jobs/189842536)

My test VM installed coverage 4.3.4, which contains in `misc.py`:
```
    287 # unittest.SkipTest doesn't exist in Python 2.6. We backport it with
    288 # unittest2 in the coverage.py test suite.  But for production, we don't need
    289 # to derive from SkipTest, so if it doesn't exist, just use Exception.
    290 
    291 class StopEverything(getattr(unittest, 'SkipTest', Exception)):
```

**Comment by tbartlett0** [comment]
Oops, committing the other forks' changes back to Master has apparently updated the pull request. (On the bright side, it seems my fix for the test failure has worked, and even py2.6 is passing now.)

If you're happy to merge all these patches then please do, otherwise let me know and I can roll back my master to just my change and submit the others separately.

**Comment by tbartlett0** [comment]
I've deleted 'have_extensions' per your comment as it's never used.

**Comment by sh0oki** [comment]
@tbartlett0 Thanks!
Can you please commit in this pull request your changes related to the specific issue they are fixing?
I agree there are other good changes to merge, but let's merge them in another pull request.
WDYT?

**Comment by tbartlett0** [comment]
Ok. This has got too messy, and as far as I can tell reverting the patches I pulled would be even more messy, so I'm going to make a new pull request on a separate branch with only my changes.

**Pull request closed without merging**